device.go

```diff
@@ -92,21 +81,35 @@ func (d *device) InstallUpdate(image io.ReadCloser, size int64) error {
 		return err
 	}
 
-	log.Debugf("Installing update to inactive partition: %s", inactivePartition)
+	b := &BlockDevice{Path: inactivePartition}
 
-	partitionSize, err := d.getPartitionSize(inactivePartition)
-	if err != nil {
+	if bsz, err := b.Size(); err != nil {
+		log.Errorf("failed to read size of block device %s: %v",
+			inactivePartition, err)
 		return err
+	} else if bsz < uint64(size) {
+		log.Errorf("update (%v bytes) is larger than the size of device %s (%v bytes)",
+			size, inactivePartition, bsz)
+		return syscall.ENOSPC
```

Comment:
Yeah, I don't think this is a problem. `syscall` is part of the binary regardless (it's used by the Go runtime code itself). Besides I think this is the standard way to return errors that aren't listed specifically in the `errors` package (and there are not many of those).